src/preact/base-element.js

```diff
@@ -177,6 +178,11 @@ export class PreactBaseElement extends AMP.BaseElement {
             'when configured with "children" or "passthrough" properties.'
         );
         this.container_ = this.element.attachShadow({mode: 'open'});
+        if (this.element.getAttribute('layout') === Layout.RESPONSIVE) {
```

Comment:
IMHO we should just always create the slot element. There's no cost if there's nothing to distribute. But I have a suggestion for a slight variation on this approach. We can support all such "service" internal elements with a single slot. It would work like this:

1. Here, we insert a single `<slot name="i-amphtml-svc">`
2. For any "service" element we decide to propagate (like sizer), all we need to do is `serviceElement.setAttribute('slot', 'i-amphtml-svc')`.